packages/api/src/period/utils/core.ts

```diff
@@ -158,6 +178,7 @@ export const findPeriodDetailsDto = async (
             $push: '$quantifications',
           },
           userAccounts: { $first: '$userAccounts' },
+          username: { $first: '$users.userName.username' },
```

Comment:
Shouldn't we try to return more "clean" useraccounts and user objects in the giver/receiver list?

The giver/receiver list is a list ofg useraccounts. Then, there are users connected to those accounts. And finally there are stats - `praiseCount` etc. 

Try this query in MongDb Compass:

```
db.praises.aggregate(
{
  $match: {}
},
{
  $lookup: {
    from: 'useraccounts',
    localField: 'receiver',
    foreignField: '_id',
    as: 'useraccount',
  },
},
{
  $unwind: "$useraccount"
},
{
  $lookup: {
    from: 'users',
    localField: 'useraccount.user',
    foreignField: '_id',
    as: 'user',
  },
},
{
  $unwind: "$user"
},
{
  $group: {
    _id: '$useraccount._id',
    userAccount: { $first: '$useraccount' },
    user: { $first: '$user' },
    praiseCount: { $count: {} },
    praiseScore: { $sum: '$scoreRealized' },
    quantifications: {
      $push: '$quantifications',
    },
  },
},
);

```

Teh output should be like:

```
{ _id: ObjectId("621f79d995a79aa51435d24e"),
  userAccount: 
   { _id: ObjectId("621f79d995a79aa51435d24e"),
     accountId: '876056250654269460',
     __v: 0,
     avatarId: '36e04ba669c77fb6873a8a862f6928f1',
     createdAt: 2022-03-02T14:06:03.220Z,
     name: 'graial#0787',
     platform: 'DISCORD',
     updatedAt: 2022-04-20T00:35:38.726Z,
     user: ObjectId("625f555a71e688919935e636"),
     activateToken: 'NEpRQZdyNsGayQ5xX9txqzg7D9BNwoHW' },
  user: 
   { _id: ObjectId("625f555a71e688919935e636"),
     __v: 0,
     createdAt: 2022-04-20T00:35:38.719Z,
     roles: [ 'USER' ],
     updatedAt: 2022-04-20T00:35:38.719Z,
     identityEthAddress: '0x05a45e7416C77C822e24e9679af29D1f07589b40',
     rewardsEthAddress: '0x05a45e7416C77C822e24e9679af29D1f07589b40',
     username: 'graial' },
  praiseCount: 22,
  praiseScore: 385.24, 
  quantifications: [...]
}
```

Reply:
I remember why, maybe @nebs-dev will agree with me, I didn't want to put all user object to response because security data, maybe it will pass some data that isn't ok for return, like nonce and etc